phylanx/execution_tree/primitives/define_function.hpp

```diff
@@ -23,8 +24,15 @@ namespace phylanx { namespace execution_tree { namespace primitives
     //
     // This is a helper primitive needed for proper binding of the expression
     // value to a variable.
-    class define_function : public primitive_component_base
+    class define_function
+        : public primitive_component_base
+        , public std::enable_shared_from_this<define_function>
     {
+    protected:
+        hpx::future<primitive_argument_type> eval(
```

Comment:
This function is left undefined, is it needed?

Reply:
Removed.